src/lib/rust.ts

```diff
@@ -285,19 +286,19 @@ const tryDownloadingBinary = async ({
 
   const url = `https://github.com/fluencelabs/marine/releases/download/${name}-v${version}/${name}-${platformToUse}-x86_64`;
 
-  try {
-    await downloadFile(binaryPath, url);
-  } catch {
-    return `Failed to download ${name}@${version} from ${url}`;
-  }
+  startSpinner(
+    `Downloading ${name}@${version} binary to ${replaceHomeDir(
+      dependencyDirPath
+    )}`
+  );
 
   try {
-    await execPromise({
-      command: "chmod",
-      args: ["+x", binaryPath],
-    });
-  } catch {
-    return `Failed to make ${name}@${version} executable by running chmod +x '${binaryPath}'`;
+    await downloadFile(binaryPath, url);
+    stopSpinner();
```

Comment:
Maybe move `stopSpinner` to `finally`?

Reply:
finally is for the cases when no matter error happened or not - you wanna do _the same thing_
here we do different things dependening on weateher error happened or not
if error happens - spinner is stopped with the message "fail"
if error does not happen - spinnner stops with the default message (in the spinner implementation that I previously used the message was "done")

so it looks like this

```
Downloading marine@0.1.1 binary to ~/.fluence/cargo/marine/0.1.1...failed
```

of


```
Downloading marine@0.1.1 binary to ~/.fluence/cargo/marine/0.1.1...done
```

but currently spinners are turned off

Reply:
ah, missed that. ok!